### PR TITLE
Nutanix gpu

### PR DESCRIPTION
**Pull request by sgaoshang**

### CODE REVIEW

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

File: ci-operator/step-registry/cucushift/installer/check/nutanix/gpu/cucushift-installer-check-nutanix-gpu-commands.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+set -o nounset
+set -o errexit
+set -o pipefail
+
+check_result=0
+
+if [[ -f ${CLUSTER_PROFILE_DIR}/secrets.sh ]]; then
+    NUTANIX_AUTH_PATH=${CLUSTER_PROFILE_DIR}/secrets.sh
+else
+    NUTANIX_AUTH_PATH=/var/run/vault/nutanix/secrets.sh
```
Comment (jhou1) [inline ci-operator/step-registry/cucushift/installer/check/nutanix/gpu/cucushift-installer-check-nutanix-gpu-commands.sh:12]:
```
Installing from release registry.build01.ci.openshift.org/ci-op-figm4jbl/release@sha256:051bcdccc9669662ba7b05dec5b4659a68f7482d02106b8c7e32b1606fe705db
/bin/bash: line 20: /var/run/secrets/ci.openshift.io/cluster-profile/pull-secret: No such file or directory
```

The error is from this line.

```
      - mountPath: /etc/pull-secret
        name: pull-secret
        readOnly: true
```

Your pull secret is mounted to this path

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-nutanix-release-4.17-e2e-nutanix

**Comment by sgaoshang** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-nutanix-release-4.17-e2e-nutanix

**Comment by sgaoshang** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-nutanix-release-4.17-e2e-nutanix

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-nutanix-ipi-gpu-f28

**Comment by sgaoshang** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-nutanix-release-4.17-e2e-nutanix

**Comment by sgaoshang** [comment]
@jianlinliu pls help to review it, thanks.

File: ci-operator/step-registry/ipi/conf/nutanix/gpu/ipi-conf-nutanix-gpu-commands.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+PATCH="${SHARED_DIR}/install-config-gpu.yaml"
+
+NUTANIX_AUTH_PATH=${CLUSTER_PROFILE_DIR}/secrets.sh
+declare gpu_name
+declare gpu_device_id
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/nutanix/gpu/ipi-conf-nutanix-gpu-commands.sh:12]: I wonder where created the var ?

Comment (sgaoshang) [reply ci-operator/step-registry/ipi/conf/nutanix/gpu/ipi-conf-nutanix-gpu-commands.sh:12]: They are in vault openshift-qe/nutanix-qe-gpu/secrets

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Pull request merged**